docs/building-modules.md

```diff
@@ -132,13 +132,80 @@ individual methods in
 
 - [DOMException](https://developer.mozilla.org/en-US/docs/Web/API/DOMException)
 
+### libp2p
+
+Zinnia comes with a built-in libp2p node based on
+[rust-libp2p](https://github.com/libp2p/rust-libp2p). The node is shared by all
+Station Modules running on Zinnia. This way we can keep the number of open
+connections lower and avoid duplicate entries in routing tables.
+
+The initial version comes with a limited subset of features. We will be adding
+more features based on feedback from our users (Station Module builders).
+
+#### Networking stack
+
+- Transport: `tcp` using system DNS resolver
+- Multistream-select V1
```

Comment:
Consider using `V1Lazy` to speed up connection establishment and stream negotiation (with the latter becoming 0RTT).

Reply:
I noticed the `V1Lazy` option in the API docs, but I am not sure if it's safe to use.

We want to allow Zinnia modules to connect to any peer running a libp2p implementation that's considered supported - at minimum the latest version of js-libp2p, go-libp2p, and rust-libp2p. We also want Zinnia modules to be able to dial any protocol, especially custom project-specific protocols.

Do you think we are unlikely to encounter unexpected issues when using `V1Lazy`?

BTW, this is exactly the most valuable kind of feedback!

Reply:
In general `V1Lazy` is interoperable with `V1`. In addition go-libp2p also supports `V1Lazy`. I am not aware of an edge case that you could encounter.

(Unless you use multistream-select to negotiate multistream-select, which I doubt you would have the need for anywhere.)

Reply:
Perfect! I opened a PR here: https://github.com/filecoin-station/zinnia/pull/87